clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -367,6 +370,162 @@ public void registeringMultipleHandlersThatTakeParamsAndBothGetTriggered() {
         assertEquals(Double.valueOf(24), value.get());
     }
 
+    @Test
+    public void checkStreamSingleItem() {
+        MockTransport mockTransport = new MockTransport();
+        HubConnection hubConnection = TestUtils.createHubConnection("http://example.com", mockTransport);
+
+        hubConnection.start().timeout(1, TimeUnit.SECONDS).blockingAwait();
+
+        AtomicBoolean completed = new AtomicBoolean();
+        AtomicBoolean onNextCalled = new AtomicBoolean();
+        Observable<String> result = hubConnection.stream(String.class, "echo", "message");
+        result.subscribe((item) -> onNextCalled.set(true), (error) -> {}
+        , () -> completed.set(true));
+
+        assertEquals("{\"type\":4,\"invocationId\":\"1\",\"target\":\"echo\",\"arguments\":[\"message\"]}" + RECORD_SEPARATOR, mockTransport.getSentMessages()[1]);
+        assertFalse(completed.get());
+        assertFalse(onNextCalled.get());
+
+        mockTransport.receiveMessage("{\"type\":2,\"invocationId\":\"1\",\"result\":\"First\"}" + RECORD_SEPARATOR);
+
+        assertTrue(onNextCalled.get());
+
+        mockTransport.receiveMessage("{\"type\":3,\"invocationId\":\"1\",\"result\":\"hello\"}" + RECORD_SEPARATOR);
+        assertTrue(completed.get());
+
+        assertEquals("First", result.timeout(1000, TimeUnit.MILLISECONDS).blockingFirst());
+    }
+
+    @Test
+    public void checkStreamMultipleItems() {
+        MockTransport mockTransport = new MockTransport();
+        HubConnection hubConnection = TestUtils.createHubConnection("http://example.com", mockTransport);
+
+        hubConnection.start().timeout(1, TimeUnit.SECONDS).blockingAwait();
+
+        AtomicBoolean completed = new AtomicBoolean();
+        Observable<String> result = hubConnection.stream(String.class, "echo", "message");
+        result.subscribe((item) -> {/*OnNext*/ },
+                (error) -> {/*OnError*/},
+                () -> {/*OnCompleted*/completed.set(true);});
+
+        assertEquals("{\"type\":4,\"invocationId\":\"1\",\"target\":\"echo\",\"arguments\":[\"message\"]}" + RECORD_SEPARATOR, mockTransport.getSentMessages()[1]);
+        assertFalse(completed.get());
+
+        mockTransport.receiveMessage("{\"type\":2,\"invocationId\":\"1\",\"result\":\"First\"}" + RECORD_SEPARATOR);
+        mockTransport.receiveMessage("{\"type\":2,\"invocationId\":\"1\",\"result\":\"Second\"}" + RECORD_SEPARATOR);
+
+        mockTransport.receiveMessage("{\"type\":3,\"invocationId\":\"1\",\"result\":\"null\"}" + RECORD_SEPARATOR);
+
+        Iterator<String> resultIterator = result.timeout(1000, TimeUnit.MILLISECONDS).blockingIterable().iterator();
+        assertEquals("First", resultIterator.next());
+        assertEquals("Second", resultIterator.next());
+        assertTrue(completed.get());
+    }
+
+    @Test
+    public void checkCancelIsSentAfterDispose() {
```

Comment:
Two more tests, one with checking a result from the completion, and one from getting an error from completion.